core/client/src/main/java/alluxio/client/file/FileSystemMasterClient.java

```diff
@@ -25,68 +24,32 @@
 import alluxio.client.file.options.MountOptions;
 import alluxio.client.file.options.SetAttributeOptions;
 import alluxio.exception.AlluxioException;
-import alluxio.thrift.AlluxioService;
-import alluxio.thrift.AlluxioTException;
-import alluxio.thrift.FileSystemMasterClientService;
-import alluxio.wire.ThriftUtils;
-
-import org.apache.thrift.TException;
 
 import java.io.IOException;
 import java.net.InetSocketAddress;
-import java.util.ArrayList;
 import java.util.List;
 
-import javax.annotation.concurrent.ThreadSafe;
-import javax.security.auth.Subject;
-
 /**
- * A wrapper for the thrift client to interact with the file system master, used by alluxio clients.
- *
- * Since thrift clients are not thread safe, this class is a wrapper to provide thread safety, and
- * to provide retries.
+ * A client to use for interacting with a file system master.
  */
-@ThreadSafe
-public final class FileSystemMasterClient extends AbstractMasterClient {
-  private FileSystemMasterClientService.Client mClient = null;
+public interface FileSystemMasterClient extends MasterClient {
```

Comment:
What is the benefit of having an interface like this? Are we expecting different implementations of this FileSystemMasterClient or are we expecting the real implementation of FileSystemMasterClient to implement multiple interfaces (including this one)?